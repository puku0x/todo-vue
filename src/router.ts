import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('./pages/todo/todo-list').then(m => m.TodoListPage)
  },
  {
    path: '/todos/new',
    component: () =>
      import('./pages/todo/todo-create').then(m => m.TodoCreatePage)
  },
  {
    path: '/todos/:id',
    component: () =>
      import('./pages/todo/todo-detail').then(m => m.TodoDetailPage)
  },
  {
    path: '/todos/:id/edit',
    component: () => import('./pages/todo/todo-edit').then(m => m.TodoEditPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
