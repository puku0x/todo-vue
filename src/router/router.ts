import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const routes: Array<RouteRecordRaw> = [
  { path: '/', redirect: '/todos' },
  {
    path: '/todos',
    component: () => import('../pages/todo').then(m => m.TodoListPage)
  }
];

export const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
});
