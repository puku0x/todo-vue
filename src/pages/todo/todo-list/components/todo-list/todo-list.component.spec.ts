import { mount } from '@vue/test-utils';
import { createRouter, createMemoryHistory } from 'vue-router';

import { Todo } from '@/models';
import { routes } from '@/router';

import TodoList from './todo-list.component.vue';

describe('TodoList', () => {
  it('render', async () => {
    const todos: Todo[] = [
      {
        id: '1',
        title: 'title',
        completed: false,
        createdAt: 123456789,
        updatedAt: 123456789
      },
      {
        id: '2',
        title: 'title',
        completed: false,
        createdAt: 123456789,
        updatedAt: 123456789
      },
      {
        id: '3',
        title: 'title',
        completed: false,
        createdAt: 123456789,
        updatedAt: 123456789
      }
    ];
    const router = createRouter({
      history: createMemoryHistory(),
      routes
    });
    router.push('/todos');
    await router.isReady();
    const wrapper = mount(TodoList, {
      global: {
        plugins: [router]
      },
      props: {
        offset: 0,
        limit: 10,
        todos
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
