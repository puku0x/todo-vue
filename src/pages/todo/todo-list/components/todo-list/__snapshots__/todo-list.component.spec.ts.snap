// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList render 1`] = `
<h2>todo-list</h2>
<p><a href="/todos/new" class="">Add a new todo</a></p>
<form> offset <input type="number" min="0" name="offset"> limit <input type="number" min="0" name="limit"></form>
<ul>
  <li><a href="/todos/1" class="">title</a></li>
  <li><a href="/todos/2" class="">title</a></li>
  <li><a href="/todos/3" class="">title</a></li>
</ul>
`;
